state/txindex/indexer.go

```diff
@@ -31,6 +31,12 @@ type TxIndexer interface {
 
 	//Set Logger
 	SetLogger(l log.Logger)
+
+	Prune(retainHeight int64) (int64, int64, error)
```

Comment:
Same comment regarding the addition of a changelog in "breaking changes"

Reply:
Done: [.changelog/unreleased/breaking-changes/1176-blockindexer-pruning.md](https://github.com/cometbft/cometbft/blob/5b2522ff35ac86a28c703071c07bf72f9bd49124/.changelog/unreleased/breaking-changes/1176-blockindexer-pruning.md)